tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/service/http/sample/HTTPCachingTestCase.java

```diff
@@ -61,4 +61,45 @@ public void testPassthroughServiceByBasePath() throws IOException, InterruptedEx
         assertEquals(response.getHeaders().get(proxyHitCount), "3");
         assertEquals(response.getData(), payload);
     }
+
+    @Test(description = "Test no-cache cache control")
+    public void testNoCacheCacheControl() throws IOException {
+        HttpResponse response = HttpClientRequest.doGet(serverInstance.getServiceURLHttp(9244, "nocache"));
+        assertEquals(response.getData(), "{\"message\":\"1st response\"}");
+        assertEquals(response.getHeaders().get(serviceHitCount), "1");
+
+        response = HttpClientRequest.doGet(serverInstance.getServiceURLHttp(9244, "nocache"));
+        assertEquals(response.getData(), "{\"message\":\"2nd response\"}");
+        assertEquals(response.getHeaders().get(serviceHitCount), "2");
+
+        response = HttpClientRequest.doGet(serverInstance.getServiceURLHttp(9244, "nocache"));
+        assertEquals(response.getData(), "{\"message\":\"2nd response\"}");
+        assertEquals(response.getHeaders().get(serviceHitCount), "3");
+    }
+
+    @Test(description = "Test max-age cache control")
+    public void testMaxAgeCacheControl() throws IOException, InterruptedException {
+        HttpResponse response = HttpClientRequest.doGet(serverInstance.getServiceURLHttp(9245, "maxAge"));
+        assertEquals(response.getData(), "public,max-age=5");
+
+        response = HttpClientRequest.doGet(serverInstance.getServiceURLHttp(9245, "maxAge"));
+        assertEquals(response.getData(), "public,max-age=5");
+
+        Thread.sleep(5000);
+
+        response = HttpClientRequest.doGet(serverInstance.getServiceURLHttp(9245, "maxAge"));
+        assertEquals(response.getData(), "{\"message\":\"after cache expiration\"}");
+    }
+
+    @Test(description = "Test must-revalidate cache control")
+    public void testMMustRevalidateCacheControl()
+            throws IOException {
+        HttpResponse response = HttpClientRequest.doGet(serverInstance.getServiceURLHttp(9247, "mustRevalidate"));
+        assertEquals(response.getData(), payload);
+        assertEquals(response.getHeaders().get(serviceHitCount), "1");
+
+        response = HttpClientRequest.doGet(serverInstance.getServiceURLHttp(9247, "mustRevalidate"));
+        assertEquals(response.getData(), payload);
+        assertEquals(response.getHeaders().get(serviceHitCount), "2");
```

Comment:
This doesn't really capture what `must-revalidate` directive does. There should be 3 requests minimum: the initial request, another request sent before the cache entry expires, a 3rd request sent after the cache entry expires. So at the end of 3 requests, the service hit count should be 2. And the proxy hit count should be 3